lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -78,7 +78,6 @@ def prune_whitelist
         batch/v1/Job
         extensions/v1beta1/DaemonSet
         extensions/v1beta1/Deployment
-        apps/v1beta1/Deployment
```

Comment:
@KnVerey do you know why we have two versions of `Deployment` in the whitelist?

Reply:
Looks like I added it back in https://github.com/Shopify/kubernetes-deploy/pull/150, which was long enough ago that the reasoning could have been based on a misunderstanding of how group versions work. I can't think of a reason that it would make sense to have to include every supported GV. If you look [here](https://github.com/kubernetes/kubernetes/blob/release-1.11/pkg/kubectl/cmd/apply.go#L501) you can see the pruning code loops over all the group versions you pass it, and for each one [runs a List call](https://github.com/kubernetes/kubernetes/blob/release-1.11/pkg/kubectl/cmd/apply.go#L587) and deletes appropriate things from that list. My guess about what's happening here is that GC is struggling in our new CI (hence needing to modify the deletion checker to look at the timestamp rather than expecting a 404), so some of the deleted resources are still there when we do the list a second time through the extra GV.

TL;DR I'm quite confident this list should only include each kind once, and that we simply need to choose a GV that is available on all k8s versions we support.